pkg/build/controller/controller.go

```diff
@@ -289,6 +291,81 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 			return fmt.Errorf("failed to update build %s/%s: %v", build.Namespace, build.Name, err)
 		}
 		glog.V(4).Infof("Build %s/%s status was updated %s -> %s", build.Namespace, build.Name, build.Status.Phase, nextStatus)
+
+		if buildutil.IsBuildComplete(build) {
+			if err := bc.executeBuildPostHooks(build); err != nil {
+				// Do not return the error, since errors executing the
+				// post hooks should not influence the build lifecycle.
+				util.HandleError(fmt.Errorf("failed to execute post hooks of build %s/%s: %v", build.Namespace, build.Name, err))
+			} else {
+				glog.V(4).Infof("Post hooks of build %s/%s were executed successfully", build.Namespace, build.Name)
+			}
+		}
+	}
```

Comment:
i'm not staking out a position per se, just raising a discussion that needs to be had so we can define what we're actually providing.

that said, i think a controller loop might be a better approach to this whole problem.  the downside to that being potential performance implications.  we can discuss on scrum.